.github/actions/deploy-environment/action.yml

```
@@ -35,3 +35,36 @@ runs:
             env:
               DOCKER_IMAGE_TAG: ${{ inputs.docker-image }}
               PR_NUMBER: ${{ inputs.pull-request-number }}
+
+          - name: Extract Terraform outputs
+            shell: bash
+            id: set_outputs
+            run: |
+              output=$(terraform -chdir=terraform/application output -json hostname)
+              echo "HOSTNAMES=$output" >> $GITHUB_ENV
+
+          - name: Set up Ruby 3.2.2
+            uses: ruby/setup-ruby@v1
+            with:
+              ruby-version: 3.2.2
+
+          - name: Bundle smoke test gems
+            shell: bash
+            run: |
+                gem install bundler
+                echo 'gem "rspec"' >> Gemfile
+                bundle
```

I think you can merge these steps by following this https://github.com/ruby/setup-ruby?tab=readme-ov-file#single-job

```suggestion
          - name: Set up Ruby 3.2.2
            uses: ruby/setup-ruby@v1
            with:
              ruby-version: 3.2.2
              bundler-cache: true
```